### PR TITLE
Tab switchingissue

### LINKED ISSUE #54: Tab Switching issues.
@iamronakshah 

Steps to be followed:
1) Open a link (https://promactoauth.azurewebsites.net)
2) Then click on various tabs continuously do it for 2 to 3 times then leave the mouse you may see the tab switching automatically.

Please see the video which I am sending you in slack personally.

### PR DESCRIPTION
**Pull request by iamronakshah**
Fixes #54 , #126 
Changed Files Are
1) HomeController.cs - Updated Index method 
2) ProjectController.cs - Added _loger to find the issue.
3) UserController.cs - Updated GetUserById Method
4) UserRoleAc.cs - Added UserRole application class.
5) ProjectRepository.cs - Added _loger to find the issue.
6) Index.cshtml - Added ViewData User Role
7) karma-test-shim.js - Added userrole for the test cases. 
8) app.component.ts - remove getRole method and use userrole model to find user role
9) app.module.ts - add userrole in app.module.ts 
10) index.html - Updated index.html
11) project-add.component.spec.ts - remove getRole method and use userrole model to find user role
12) userrole.model.ts - added userrole model
13) user-details.component.ts - removed getrole method and use userrole model to find user role
14) user-edit.component.ts -removed getrole method and use userrole model to find user role
15) user.component.ts - removed getrole method and use userrole model to find user role
16) user.service.ts  - Updated user.service.ts
15)project-list.component.spec.ts - Added user role
16) user-details.component.spec.ts - Added user role

### CODE REVIEW

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs [hunk 1]
```diff
@@ -126,13 +126,13 @@ public IActionResult GetRole()
 
 
         [HttpGet]
-        [Route("{id}")]
+        [Route("getUserById/{userId}")]
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs]: We don't use this naming conventions

File: Promact.Oauth.Server/src/Promact.Oauth.Server/wwwroot/app/users/user-edit/user-edit.component.ts [hunk 2]
```diff
@@ -66,43 +70,15 @@ export class UserEditComponent {
 
         }, err => {
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/wwwroot/app/users/user-edit/user-edit.component.ts]: what happens when there is error? what will you show user?

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs (hunk 1 above)
Comment (iamronakshah) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs]: yes sir i know but this was changed before your mail. but now i will update it.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/wwwroot/app/users/user-edit/user-edit.component.ts (hunk 2 above)
Comment (iamronakshah) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/wwwroot/app/users/user-edit/user-edit.component.ts]: now I have added console.log(err.statusText) for the showing error. 
I think it not for the end user.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ProjectController.cs [hunk 3]
```diff
@@ -60,19 +64,23 @@ public ProjectController(PromactOauthDbContext appContext, IProjectRepository pr
...
                 {
+                    _logger.LogInformation("call project repository for projects");
                     return await _projectRepository.GetAllProjects();
                 }
             }
             catch (Exception ex)
             {
-                ex.ToExceptionless().Submit();
-                throw ex;
+               _logger.LogError("Exception " + ex.ToString());
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ProjectController.cs]: There is no need to explicitly  send the exception to exceptionless. Also, no need to write exception to logger.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ProjectController.cs [hunk 4]
```diff
@@ -60,19 +64,23 @@ public ProjectController(PromactOauthDbContext appContext, IProjectRepository pr
...
+                    _logger.LogInformation("call project repository for User");
                     return await _projectRepository.GetAllProjectForUser(user.Id);
                 }
                 else
                 {
+                    _logger.LogInformation("call project repository for projects");
                     return await _projectRepository.GetAllProjects();
                 }
             }
             catch (Exception ex)
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ProjectController.cs]: I think I have conveyed this that remove all try.. catch unless you are catching any specific exception

File: Promact.Oauth.Server/src/Promact.Oauth.Server/wwwroot/app/users/user-edit/user-edit.component.ts (hunk 2 above)
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/wwwroot/app/users/user-edit/user-edit.component.ts]: The method is for edit user. What happens if edit fails for any reason. You are not providing any feedback to a user that it failed. I think there needs to be a better way to handle the error and report it to client.

Comment (iamronakshah) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/wwwroot/app/users/user-edit/user-edit.component.ts]: now i have added toast for displaying message for the end users(client).

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ProjectController.cs (hunk 4 above)
Comment (iamronakshah) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ProjectController.cs]: I have removed try.. catch unless I catch specific exception but that is in my new branch, reflector project controller. if you want for this branch then i will update it in this branch.

**Pull request merged**